packages/credentials/src/V1/KiltCredentialV1.ts

```diff
@@ -327,15 +328,76 @@ export function fromInput({
 
 const cachingCTypeLoader = newCachingCTypeLoader()
 
+// Helper function to check if CType is nested
+function cTypeTypeFinder(cType: ICType): boolean {
+  function hasRef(obj: any): boolean {
+    if (typeof obj !== 'object' || obj === null) return false
+    
+    if ('$ref' in obj) return true
+    
+    return Object.values(obj).some(value => {
+      if (Array.isArray(value)) {
+        return value.some(item => hasRef(item))
+      }
+      if (typeof value === 'object') {
+        return hasRef(value)
+      }
+      return false
+    })
+  }
+  return hasRef(cType.properties)
+}
```

Comment:
No need to make another `Named` function inside this function. 

```suggestion
// Helper function to check if CType is nested
function cTypeTypeFinder(cType: ICType): boolean {

    if (typeof cType.properties !== 'object' || cType.properties === null) return false
    
    if ('$ref' in cType.properties) return true
    
    return Object.values(cType.properties).some(value => {
      if (Array.isArray(value)) {
        return value.some(item => hasRef(item))
      }
      if (typeof value === 'object') {
        return hasRef(value)
      }
      return false
    })
}
```

Reply:
I understand that you're defining a function to call it recursively, but can't you just do that with the outer function directly? Also, more generally, I think all that you need is the second function; if you're already recursing through the CType, collect all the references, we _will_ need them eventually.

Reply:
Fixed at: https://github.com/KILTprotocol/sdk-js/pull/919/commits/77b9c3d89f213c51b6b428ade650cc04c50efc2b